### PR TITLE
Replace ReSpec HTML drafts with Markdown versions

### LINKED ISSUE #554: Change specification source from Respec to Markdown
I think we should change the OCFL specifications from Respec to Markdown:
  * In Markdown we'd have better control of formatting and not end up fighting Respec (see [issues and PRs](https://github.com/OCFL/spec/issues?q=respec+))
  * We would immediately solve #505 with the ability to add `ocfl.md` and/or `ocfl.html` to the Storage Root
  * Compilation with Jekyll is required (a downside) but then we would have a static HTML file that doesn't rely on JavaScript lookups for rendering (a big upside, meaning our spec is preservable)
  * We don't need the "looks like a W3C spec" fake authenticity

### PR DESCRIPTION
**Pull request by zimeon**
Fixes #554

There is the following fixup required:

* in spec, rendered version: https://zimeon.github.io/ocfl-spec/draft/spec/
  * Indentation in example in 3.5.3.1 Version is not right: `empty.txt` and `empty2.txt` should be at some level
  * Non-normative notes that span multiple paragraphs need a tweak to avoid the quote bar having breaks in it
  * Adjust wrapping in RFC2119 reference so that the markdown doesn't incorrectly pick it up `1997.` as the start of a numbered list.
* in implementation notes, rendered version: https://zimeon.github.io/ocfl-spec/draft/implementation-notes/
  * In section 2.2, move examples to after the descriptions of Flat, PairTree and Truncated

If we have approval to go ahead (and commitment not to merge anything else in the meantime) that I can do these and get re-approval before merge. Must be careful to do a squash&merge

### CODE REVIEW

**Review by neilsjefferies — APPROVED** [review]

**Comment by awoods** [comment]
Should we also remove: [respec-w3c-common.js](https://github.com/OCFL/spec/blob/main/draft/respec-w3c-common.js)?

**Comment by awoods** [comment]
I am happy to move the current PR (which is fantastic) forward and then iterate on the refinements. Potential such refinements, in addition to those noted in the PR description:
1. The additional title at the top of the document seems unnecessary
2. Is there a way of having the 'table of contents' be in a side panel, so that it is always available.

Thanks!

**Review by awoods — CHANGES_REQUESTED** [review]
Approved, pending response minor comments.

**Comment by zimeon** [comment]
> Should we also remove: [draft/respec-w3c-common.js](https://github.com/OCFL/spec/blob/main/draft/respec-w3c-common.js)?

Yes, done in https://github.com/OCFL/spec/pull/596/commits/bc8b159cae0581637b0d9d7cf70128b1e552f077

> I am happy to move the current PR (which is fantastic) forward and then iterate on the refinements. Potential such refinements, in addition to those noted in the PR description:
> 
> 1. The additional title at the top of the document seems unnecessary

We use this "extra" title (actually the `site.title`) on the homepage and news page. I have added into `_layouts/default.html` a check for an extra page variable `page.no_site_title` which can be used in the front matter to disable the extra title, this is added as `no_site_title: true` in both spec and implementation notes.

> 2. Is there a way of having the 'table of contents' be in a side panel, so that it is always available.

Yes, may take some work to get it right but this is the case for IIIF specs. See, e.g. https://iiif.io/api/presentation/3.0/

File: draft/implementation-notes/index.md [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+### 1.1 Rebuildability
+{: #rebuildability}
+
+A key goal of the OCFL is the rebuildability of a repository from an OCFL storage root without additional information
+resources. Consequently, a key implementation consideration should be to ensure that OCFL objects contain all the data
+and metadata required to achieve this. With reference to the \[[OAIS](#ref-oais)\] model, this would include all the
+descriptive, administrative, structural, representation, and preservation metadata relevant to the object.
+
+Additionally, as an aid to those who may need to recover OCFL objects in the future, it is recommended that a copy of
+the \[[OCFL-Specification](#ref-ocfl-specification)\] is stored in the top level of the OCFL storage root. The OCFL
```
[the remark this answers is not in the document]
Comment (zimeon) [reply draft/implementation-notes/index.md:66]: We perhaps need to write a style guide. For this PR I made things look like the ReSpec version

**Review by awoods — APPROVED** [review]

**Pull request merged**